### PR TITLE
Project search assets

### PR DESCRIPTION
**Pull request by kbirk**

### CODE REVIEW

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/elasticsearch/ElasticsearchService.java [hunk 1]
```diff
@@ -471,6 +471,34 @@ public <T, Partial> void update(final String index, final String id, final Parti
 		}
 	}
 
+	/**
+	 * Update a document from an index.
+	 *
+	 * @param index The index to remove the document from
+	 * @param id    The id of the document to remove
```
Comment (YohannParis) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/elasticsearch/ElasticsearchService.java:478]: Your JavaDoc is missing `partial` and `routing`.

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/ProjectSearchService.java [hunk 2]
```diff
@@ -205,10 +327,203 @@ public List<ProjectDocument> searchProjectsForUser(
...
+									index = i;
+									break;
+								}
+							}
+							if (index != -1) {
+								projectDoc.getAssetEmbeddings().set(index, projectAssetEmbedding);
+							} else {
+								projectDoc.getAssetEmbeddings().add(projectAssetEmbedding);
+							}
+						}
```
Comment (YohannParis) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/ProjectSearchService.java:386]: Why not deleting the embeddings and adding the new generated ones?
I'm not sure I understand, maybe it's an ES logic to do it this way.

**Review by YohannParis — APPROVED** [review]
LGTM

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/TerariumAssetServiceWithSearch.java [hunk 3]
```diff
@@ -364,4 +380,39 @@ public void updateIndexInPlace() throws IOException {
 		// refresh the index
 		elasticService.refreshIndex(getAssetAlias());
 	}
+
+	protected boolean isRunningTestProfile() {
```
Comment (dvince2) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/TerariumAssetServiceWithSearch.java:384]: I'm starting to see this method copied and pasted around more and more (I just had to use it for checking test with the DKG import). I wonder if we should centralize this....

**Review by dvince2 — APPROVED** [review]

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/elasticsearch/ElasticsearchService.java (hunk 1 above)
Comment (kbirk) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/elasticsearch/ElasticsearchService.java:510]: Done

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/ProjectSearchService.java (hunk 2 above)
Comment (kbirk) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/ProjectSearchService.java:387]: The embeddings are part of a larger document. So we update them in place and overwrite the entire document.

**Pull request merged**